src/piadm.sh

```diff
@@ -724,30 +719,31 @@ install_pi_CN() {
 	# For now, use bootparams to get the URL needed, and pull
 	# files from there.  If there's a better way to obtain things, use it.
 	unix_path=$(bootparams | grep boot-file | awk -F= '{print $2}')
-	archive_prefix=$(echo $unix_path | sed 's/kernel\/amd64\/unix/amd64/g')
+	archive_prefix=$(echo "$unix_path" | sed 's/kernel\/amd64\/unix/amd64/g')
 
 	# Reality check the buildstamp passed, which will become installstamp,
 	# is in the unix_path.
-	echo $unix_path | grep -q $1 || corrupt "PI $1 not in $unix_path"
+	echo "$unix_path" | grep -q "$1" || corrupt "PI $1 not in" "$unix_path"
```

Comment:
Cleaner:

    grep -q "$1" <<< "$unix_path" || corrupt "PI $1 is not in $unix_path"

Reply:
Same here, holding off.